internal/training/grpo_engine/grpo.c

```
@@ -0,0 +1,471 @@
+    return grpo_policy_prefill_internal(ctx->policy, prompt, prompt_len);
+}
+
+int grpo_generate_continue(GrpoCtx *ctx, int *output, int max_len,
+                          float *logprobs_out, float temp, float top_p) {
+    if (!ctx || !ctx->policy || !output) return -1;
+    return grpo_policy_generate_continue_internal(ctx->policy, output, max_len,
+                                                  logprobs_out, temp, top_p,
+                                                  &ctx->rng_state);
+}
+
+void grpo_save_kv_snapshot(GrpoCtx *ctx) {
```

Cppcheck (2.21.0)

[style] The function 'grpo_save_kv_snapshot' is never used.

(unusedFunction)